static/js/components/inputs/DateField.js

```diff
@@ -184,6 +186,7 @@ export default class DateField extends React.Component {
         fullWidth={true}
         value={edit.year !== undefined ? edit.year : ""}
         onChange={e => setNewDate(undefined, undefined, e.target.value)}
+        onBlur={onBlur}
```

Comment:
yeah, that's exactly why I ended up deciding not to, I couldn't figure out a good way to keep the validation errors from appearing until after the user had finished entering a date or had clearly abandoned it.